src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq.c

```diff
@@ -1677,6 +1680,111 @@ static int vmr_status_query(struct platform_device *pdev)
 	return vmr_control_op(pdev, XGQ_CMD_VMR_QUERY);
 }
 
+static void clk_scaling_cq_result_copy(struct xocl_xgq_vmr *xgq,
+                                       struct xocl_xgq_vmr_cmd *cmd)
+{
+	struct xgq_cmd_cq_default_payload *payload =
+		(struct xgq_cmd_cq_default_payload *)&cmd->xgq_cmd_cq_payload;
+	struct xgq_cmd_cq_clk_scaling_payload *cs_payload=
+		(struct xgq_cmd_cq_clk_scaling_payload *)&xgq->xgq_cq_payload;
+
+	mutex_lock(&xgq->xgq_lock);
+	memcpy(&xgq->xgq_cq_payload, payload, sizeof(*payload));
+	xgq->pwr_scaling_ovrd_limit = cs_payload->pwr_scaling_limit;
+	xgq->temp_scaling_ovrd_limit = cs_payload->temp_scaling_limit;
+	mutex_unlock(&xgq->xgq_lock);
+}
+
+static int clk_scaling_configure_op(struct platform_device *pdev,
+                                    enum xgq_cmd_clk_scaling_app_id aid,
+                                    bool enable, uint16_t pwr_ovrd_limit,
+                                    uint8_t temp_ovrd_limit)
+{
+	struct xocl_xgq_vmr *xgq = platform_get_drvdata(pdev);
+	struct xocl_xgq_vmr_cmd *cmd = NULL;
+	struct xgq_cmd_clk_scaling_payload *payload = NULL;
+	struct xgq_cmd_sq_hdr *hdr = NULL;
+	int ret = 0;
+	int id = 0;
+
+	if (xgq->xgq_halted) {
+		XGQ_WARN(xgq, "VMR XGQ service is haulted. skip.");
+		return -EIO;
+	}
+
+	cmd = kmalloc(sizeof(*cmd), GFP_KERNEL);
+	if (!cmd) {
+		XGQ_ERR(xgq, "kmalloc failed, retry");
+		return -ENOMEM;
+	}
+
+	memset(cmd, 0, sizeof(*cmd));
+	cmd->xgq_cmd_cb = xgq_complete_cb;
+	cmd->xgq_cmd_arg = cmd;
+	cmd->xgq_vmr = xgq;
+
+	payload = &(cmd->xgq_cmd_entry.clk_scaling_payload);
+	payload->aid = aid;
+	if (aid == XGQ_CMD_CLK_THROTTLING_AID_CONFIGURE)
+	{
+		payload->scaling_en = enable ? 1 : 0;
+		if (pwr_ovrd_limit)
```

Comment:
The logic here has a different logic then expected.   Is the intention truly:
1. If the pwr_ovrd_limit is zero, honor the previous value of psr_scaling_ovrd_limit.
2. If the pwr_ovrd_limit is non-zero, over the previous value of psr_scaling_ovrd_limt with this one.

If so, another possible way of writing this code could be:
`   payload->psr_scaling_ovrd_limit = psr_ovrd_limit ? psr_ovr_limit :  payload->psr_scaling_ovrd_limit`

If not, the why couldn't the code be writing as follows:
`   payload->psr_scaling_ovrd_limit = psr_ovrd_limit`

Reply:
payload will have clk throttling feature enable, and override limits. Intention here is to send the values only if override limits are valid (non zeros).